From 9df3cb58a19f38157a615c311356cfd6ae55d2d7 Mon Sep 17 00:00:00 2001
From: Pavel Valena <pvalena@redhat.com>
Date: Sun, 23 Jul 2023 19:44:17 +0200
Subject: [PATCH 23/23] fix(rngd): install system service file

as there's no reason to keep a copy; there shouldn't be any modifications.

In case there are args stored in a separate file (Fedora and alike),
it needs to be supplied too, but without the option to change the user.
---
 modules.d/06rngd/module-setup.sh | 7 ++++++-
 modules.d/06rngd/rngd.service    | 8 --------
 modules.d/06rngd/sysconfig       | 1 +
 3 files changed, 7 insertions(+), 9 deletions(-)
 delete mode 100644 modules.d/06rngd/rngd.service
 create mode 100644 modules.d/06rngd/sysconfig

diff --git a/modules.d/06rngd/module-setup.sh b/modules.d/06rngd/module-setup.sh
index aec8d576..e8bdf7f5 100755
--- a/modules.d/06rngd/module-setup.sh
+++ b/modules.d/06rngd/module-setup.sh
@@ -32,7 +32,12 @@ depends() {
 
 install() {
     inst rngd
-    inst_simple "${moddir}/rngd.service" "${systemdsystemunitdir}/rngd.service"
+    inst_simple "${systemdsystemunitdir}/rngd.service"
+
+    if [ -r /etc/sysconfig/rngd ]; then
+        inst_simple "${moddir}/sysconfig" "/etc/sysconfig/rngd"
+    fi
+
     # make sure dependent libs are installed too
     inst_libdir_file opensc-pkcs11.so
 
diff --git a/modules.d/06rngd/rngd.service b/modules.d/06rngd/rngd.service
deleted file mode 100644
index dd5374d7..00000000
--- a/modules.d/06rngd/rngd.service
+++ /dev/null
@@ -1,8 +0,0 @@
-[Unit]
-Description=Hardware RNG Entropy Gatherer Daemon
-DefaultDependencies=no
-Before=systemd-udevd.service
-ConditionVirtualization=!container
-
-[Service]
-ExecStart=/usr/sbin/rngd -f
diff --git a/modules.d/06rngd/sysconfig b/modules.d/06rngd/sysconfig
new file mode 100644
index 00000000..68047ec1
--- /dev/null
+++ b/modules.d/06rngd/sysconfig
@@ -0,0 +1 @@
+RNGD_ARGS="-x pkcs11 -x nist"
-- 
2.42.0

